[CI] DatafeedJobsRestIT class failing

**Build Scans:**
- [elasticsearch-periodic #5168 / openjdk21_checkpart4_java-fips-matrix](https://gradle-enterprise.elastic.co/s/cpqzyf3llnjdq)
- [elasticsearch-pull-request #44085 / part-4](https://gradle-enterprise.elastic.co/s/6p5gnffnemybe)
- [elasticsearch-pull-request #43932 / part-4](https://gradle-enterprise.elastic.co/s/vzkhnvq7hermu)
- [elasticsearch-pull-request #43505 / part-4](https://gradle-enterprise.elastic.co/s/67h222r5mzgwu)
- [elasticsearch-pull-request #43443 / part-4](https://gradle-enterprise.elastic.co/s/ieqwp6colwktw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ml:qa:native-multi-node-tests:javaRestTest" --tests "org.elasticsearch.xpack.ml.integration.DatafeedJobsRestIT.testLookbackOnlyWithRuntimeFields" -Dtests.seed=E28CDC0671B1A13E -Dtests.locale=naq -Dtests.timezone=America/Montevideo -Druntime.java=21 -Dtests.fips.enabled=true
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ml.integration.DatafeedJobsRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.xpack.ml.integration.DatafeedJobsRestIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 2 active tasks found:
cluster:admin/xpack/ml/job/close    z-Mju6L3TEyJVV1HCrOduA:24555 -                            transport  1733147134469 13:45:34 1.8m        127.0.0.1 javaRestTest-2 
cluster:admin/xpack/ml/job/close[n] z-Mju6L3TEyJVV1HCrOduA:24561 z-Mju6L3TEyJVV1HCrOduA:24555 transport  1733147134477 13:45:34 1.8m        127.0.0.1 javaRestTest-2 
 expected:<0> but was:<2>
```

**Issue Reasons:**
- [main] 5 failures in class org.elasticsearch.xpack.ml.integration.DatafeedJobsRestIT (0.7% fail rate in 740 executions)
- [main] 4 failures in step part-4 (1.5% fail rate in 263 executions)
- [main] 4 failures in pipeline elasticsearch-pull-request (1.5% fail rate in 261 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -84,9 +84,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/115816
 - class: org.elasticsearch.xpack.application.connector.ConnectorIndexServiceTests
   issue: https://github.com/elastic/elasticsearch/issues/116087
-- class: org.elasticsearch.xpack.ml.integration.DatafeedJobsRestIT
-  method: testLookbackWithIndicesOptions
-  issue: https://github.com/elastic/elasticsearch/issues/116127
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=transform/transforms_start_stop/Test start already started transform}
   issue: https://github.com/elastic/elasticsearch/issues/98802
@@ -152,8 +149,6 @@ tests:
 - class: org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT
   method: test {p0=search.highlight/50_synthetic_source/text multi unified from vectors}
   issue: https://github.com/elastic/elasticsearch/issues/117815
-- class: org.elasticsearch.xpack.ml.integration.DatafeedJobsRestIT
-  issue: https://github.com/elastic/elasticsearch/issues/111319
 - class: org.elasticsearch.xpack.esql.plugin.ClusterRequestTests
   method: testFallbackIndicesOptions
   issue: https://github.com/elastic/elasticsearch/issues/117937

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsRestIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsRestIT.java
@@ -37,7 +37,6 @@ import java.util.stream.Collectors;
 import static org.hamcrest.Matchers.anyOf;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.not;
 
 public class DatafeedJobsRestIT extends ESRestTestCase {
 

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsRestIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsRestIT.java
@@ -503,12 +503,12 @@ public class DatafeedJobsRestIT extends ESRestTestCase {
             }""");
         client().performRequest(createJobRequest);
         String datafeedId = jobId + "-datafeed";
-        new DatafeedBuilder(datafeedId, jobId, "*hidden-*").setIndicesOptions("""
+        new DatafeedBuilder(datafeedId, jobId, "hidden-*").setIndicesOptions("""
             {"expand_wildcards": ["all"],"allow_no_indices": true}""").build();
 
         StringBuilder bulk = new StringBuilder();
 
-        Request createGeoData = new Request("PUT", "/.hidden-index");
+        Request createGeoData = new Request("PUT", "/hidden-index");
         createGeoData.setJsonEntity("""
             {
               "mappings": {
@@ -528,23 +528,23 @@ public class DatafeedJobsRestIT extends ESRestTestCase {
         client().performRequest(createGeoData);
 
         bulk.append("""
-            {"index": {"_index": ".hidden-index", "_id": 1}}
+            {"index": {"_index": "hidden-index", "_id": 1}}
             {"time":"2016-06-01T00:00:00Z","value": 1000}
-            {"index": {"_index": ".hidden-index", "_id": 2}}
+            {"index": {"_index": "hidden-index", "_id": 2}}
             {"time":"2016-06-01T00:05:00Z","value":1500}
-            {"index": {"_index": ".hidden-index", "_id": 3}}
+            {"index": {"_index": "hidden-index", "_id": 3}}
             {"time":"2016-06-01T00:10:00Z","value":1600}
-            {"index": {"_index": ".hidden-index", "_id": 4}}
+            {"index": {"_index": "hidden-index", "_id": 4}}
             {"time":"2016-06-01T00:15:00Z","value":100}
-            {"index": {"_index": ".hidden-index", "_id": 5}}
+            {"index": {"_index": "hidden-index", "_id": 5}}
             {"time":"2016-06-01T00:20:00Z","value":1}
-            {"index": {"_index": ".hidden-index", "_id": 6}}
+            {"index": {"_index": "hidden-index", "_id": 6}}
             {"time":"2016-06-01T00:25:00Z","value":1500}
-            {"index": {"_index": ".hidden-index", "_id": 7}}
+            {"index": {"_index": "hidden-index", "_id": 7}}
             {"time":"2016-06-01T00:30:00Z","value":1500}
-            {"index": {"_index": ".hidden-index", "_id": 8}}
+            {"index": {"_index": "hidden-index", "_id": 8}}
             {"time":"2016-06-01T00:40:00Z","value":2100}
-            {"index": {"_index": ".hidden-index", "_id": 9}}
+            {"index": {"_index": "hidden-index", "_id": 9}}
             {"time":"2016-06-01T00:41:00Z","value":0}
             """);
         bulkIndex(bulk.toString());
@@ -1802,7 +1802,7 @@ public class DatafeedJobsRestIT extends ESRestTestCase {
         bulkRequest.addParameter("refresh", "true");
         bulkRequest.addParameter("pretty", null);
         String bulkResponse = EntityUtils.toString(client().performRequest(bulkRequest).getEntity());
-        assertThat(bulkResponse, not(containsString("\"errors\": false")));
+        assertThat(bulkResponse, containsString("\"errors\" : false"));
     }
 
     private Response createJobAndDataFeed(String jobId, String datafeedId) throws IOException {

